pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -211,53 +217,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return r.result(err).requeue(duration).ret()
 		}
 		if !workflow.IsFailedAfterRetry(app) || !feature.DefaultMutableFeatureGate.Enabled(features.EnableSuspendOnFailure) {
 			r.stateKeep(logCtx, handler, app)
 		}
-		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false, true)
-	case common.WorkflowStateTerminated:
+		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false, isUpdate)
+	case workflowv1alpha1.WorkflowRunTerminated:
 		logCtx.Info("Workflow return state=Terminated")
 		handler.UpdateApplicationRevisionStatus(logCtx, handler.latestAppRev, false, app.Status.Workflow)
-		if err := r.doWorkflowFinish(app, wf, workflowState); err != nil {
-			return r.endWithNegativeCondition(ctx, app, condition.ErrorCondition(common.WorkflowCondition.String(), errors.WithMessage(err, "DoWorkflowFinish")), common.ApplicationRunningWorkflow)
-		}
-		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowTerminated, false, true)
-	case common.WorkflowStateExecuting:
+		r.doWorkflowFinish(app, workflowState)
+		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowTerminated, false, isUpdate)
+	case workflowv1alpha1.WorkflowRunExecuting:
 		logCtx.Info("Workflow return state=Executing")
-		_, err = r.gcResourceTrackers(logCtx, handler, common.ApplicationRunningWorkflow, false, true)
-		return r.result(err).requeue(wf.GetBackoffWaitTime()).ret()
-	case common.WorkflowStateSucceeded:
+		_, err = r.gcResourceTrackers(logCtx, handler, common.ApplicationRunningWorkflow, false, isUpdate)
+		return r.result(err).requeue(executor.GetBackoffWaitTime()).ret()
+	case workflowv1alpha1.WorkflowRunSucceeded:
 		logCtx.Info("Workflow return state=Succeeded")
 		handler.UpdateApplicationRevisionStatus(logCtx, handler.currentAppRev, true, app.Status.Workflow)
-		if err := r.doWorkflowFinish(app, wf, workflowState); err != nil {
-			return r.endWithNegativeCondition(logCtx, app, condition.ErrorCondition(common.WorkflowCondition.String(), errors.WithMessage(err, "DoWorkflowFinish")), common.ApplicationRunningWorkflow)
-		}
+		r.doWorkflowFinish(app, workflowState)
 		app.Status.SetConditions(condition.ReadyCondition(common.WorkflowCondition.String()))
 		r.Recorder.Event(app, event.Normal(velatypes.ReasonApplied, velatypes.MessageWorkflowFinished))
 		logCtx.Info("Application manifests has applied by workflow successfully")
 		if !EnableReconcileLoopReduction {
-			return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowFinished, false, true)
-		}
-	case common.WorkflowStateFinished:
-		logCtx.Info("Workflow state=Finished")
-		if status := app.Status.Workflow; status != nil && status.Terminated {
-			return r.result(nil).ret()
+			if result, err := r.gcResourceTrackers(logCtx, handler, common.ApplicationRunning, false, isUpdate); err != nil {
+				return result, err
+			}
 		}
-	case common.WorkflowStateSkipping:
+	case workflowv1alpha1.WorkflowRunSkipped:
```

Comment:
we shouldn't use `WorkflowRun` as a prefix here. We need a common state such as "WorkflowSkipped"

Reply:
the same on above